rpcs3/Emu/Cell/lv2/sys_rsx.cpp

```diff
@@ -58,8 +62,20 @@ s32 sys_rsx_memory_allocate(vm::ptr<u32> mem_handle, vm::ptr<u64> mem_addr, u32
  */
 s32 sys_rsx_memory_free(u32 mem_handle)
 {
-	sys_rsx.todo("sys_rsx_memory_free(mem_handle=0x%x)", mem_handle);
+	sys_rsx.warning("sys_rsx_memory_free(mem_handle=0x%x)", mem_handle);
+
+	if (!vm::check_addr(0xC0000000))
+	{
+		return CELL_EINVAL;
+	}
 
+	if (fxm::get_always<SysRsxConfig>()->context_base)
+	{
+		// TODO: What happens here?
```

Comment:
Likely leads to a crash or a GPU reset anyway, so throwing is 'correct' imo. The onus here would be on the caller to make sure it doesn't happen.